### DISCUSSION TITLE
How to send api request from local UI to Planetary Hub to get the results?

### DISCUSSION THREAD

**Discussion by shriya96** [microsoft/PlanetaryComputer#236] [Ideas]
HI, i am trying to connect my Local UI with a py file in planetary computer hub using Fast api to fetch the data and maps generated, however i am unable to configure the url. Let me know if this is possible in planetary computer hub

**Comment by TomAugspurger** [comment]
Hi @shriya96. I'm not sure if that's possible. I believe that [JupyterHub's](https://jupyterhub.readthedocs.io/en/stable/reference/rest-api.html#/jupyterhub-rest-api) REST API is mostly used for managing notebook servers, and not for managing the contents of the volumes within those servers.

By the way, [we don't recommend](https://planetarycomputer.microsoft.com/docs/overview/environment/#understanding-the-file-system) your home directory on the Hub for storing outputs, at least not large ones. It's usually better to store those in Blob Storage. See https://planetarycomputer.microsoft.com/docs/quickstarts/storage/ for some more.

**Reply by shriya96** [reply]
Thanks for the quick response.

I need to create a front end using React (or Streamlit) to be hosted on Azure App service. In the front end the user will make some selections on the dataset, climate model etc. 

Is it possible to use STAC API or Data API to get data and do the computations in Planetary computer and then show the result (response of API) as a Plot in the front end?

If so how can I achieve it?

**Reply by TomAugspurger** [reply]
> Is it possible to use STAC API or Data API to get data and do the computations in Planetary computer and then show the result (response of API) as a Plot in the front end?

Depending on the computation, yes. That's exactly how our [Explorer](https://planetarycomputer.microsoft.com/explore) tool works. If you look at the requests it's making, you'll see a bunch of requests to our STAC and Data APIs.

The Data API builds on [TiTiler](https://developmentseed.org/titiler/). It uses [rio-tiler](https://cogeotiff.github.io/rio-tiler/) for some of the computation expressions.
